### PR TITLE
i18n(it): added translation to "update  to v2"

### PR DESCRIPTION
**Pull request by akaik0**
#### What kind of changes does this PR include?

- Translated content

#### Description

- Closes # 
- What does this PR change? Give us a brief description.
- Did you change something visual? A before/after screenshot can be helpful.

This PR contains the translation to italian of the page upgarde-to-v2

### CODE REVIEW

**Comment by dreyfus92** [comment]
@akaik0 just a small suggestion, since I can't review `italian` cause i don't know it. Try changing the title of the PR to 
`i18n(it): w/e you did` so it can be easier to look up for other translators.

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+
+### Rimosso: Supporto di Node 14
+
+È previsto che nell'Aprile 2023 Node 14 concluderà il suo ciclo di vita.
+
+Astro v2.0 abbandona il supporto di Node 14 interamente, affinchè gli utenti di Astro possano trarre vantaggio dalle funzionalità più moderne di Node.
+
+#### Cosa devo fare?
+
+Controlla che sia l'ambiente di sviluppo che l'ambiente di distribuzione stiano usando **Node `16.12.0` o versini successive **.
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:62]: It should read `versioni`, instead of `versini`

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+  
+  # Example: upgrade React and Tailwind integrations
+  yarn add @astrojs/react@latest @astrojs/tailwind@latest
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+## Astro v2.0 Cambiamenti principali
+
+Astro v2.0 include alcuni cambiamenti importanti, come anche la rimozione di funzionalità obsolete. Se il tuo progetto non fnziona come dovrebbe dopo l'aggiornamento alla v2.0, consulta questa guida per una panoramica di tutti i cambiamenti e le istruzioni su come aggiornare il tuo codice.
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:50]: **typo:** It should read `funziona`, not `fnziona`

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+
+Vedi il [registro delle modifiche](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md) per le note di rilascio complete.
+
+### Rimosso: Supporto di Node 14
+
+È previsto che nell'Aprile 2023 Node 14 concluderà il suo ciclo di vita.
+
+Astro v2.0 abbandona il supporto di Node 14 interamente, affinchè gli utenti di Astro possano trarre vantaggio dalle funzionalità più moderne di Node.
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:58]: **nitpick:** I would rephrase this as `Astro v2.0 non supporterà più Node 14, in modo che gli utenti` ...

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+   Puoi specificare Node `16.12.0` per il tuo progetto Astro sia in un'impostazione della configurazione del pannello di controllo che in un file `.nvmrc`.
+
+
+### Riservato: `src/content/`
+
+Astro v2.0 ora include l'API delle Collezioni per organizzare i tuoi file Markdown e MDX in [raccolte di contenuti](/it/guides/content-collections/). Questa API riserva `src/content/` come cartella speciale.
+
+#### Cosa devo fare?
+
+Cambia il nome alla cartella esistente `src/content/` per evitare conflitti.Questa cartella, se esiste, ora può essere utilizzata solo per [raccolte di contenuti](/it/guides/content-collections/).
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:84]: **typo:** Add a blank between `conflitti.` and `Questa`

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+
+### Cambiato: Configurazione del plugin MDX
+
+#### Sostituito: `extendPlugins` cambiato con `extendMarkdownConfig`
+
+Nella v1.x, l'opzione `extendPlugins` dell'integrazione MDX gestiva come i tuoi file MDX dovevano ereditare le configurazioni di Markdown: tutte le tue configurazioni di Markdown (`markdown`), o solamente i plugin predefiniti di Astro (`default`).
+
+Astro v2.0 sostituisce questo comportamento controllato da `mdx.extendPlugins` con tre nuove opzioni, configurabili indipendentemente, che sono impostate a `true`:
+
+- **[`mdx.extendMarkdownConfig`](/it/guides/integrations-guide/mdx/#extendmarkdownconfig)** to inherit all or none of your Markdown configuration
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:184]: **issue:** These 3 lines are still in English, e.g. `to inherit`.
I would suggest:
Line 184: `per ereditare tutte, o nessuna, le tue configurazioni Markdown`
Line 185: `per abilitare o disabilitare GitHub-Flavored Markdown nei file MDX`
Line 186: `per abilitare o disabilitare SmartyPants nei file MDX`

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+
+Nella v1.x, potevi scegliere se disabilitare entrambi i plugin di Markdown di Astro predefiniti (GitHub-Flavored Markdown e SmartyPants) impostando `markdown.extendDefaultPlugins: false`.
+
+Astro v2.0 sostituisce `markdown.extendDefaultPlugins: false` con delle opzioni booleane separate per controllare indipendentemente ognuno dei plugin Markdown di Astro. Questi sono abilitati di default e possono essere impostati su `false` in modo indipendente.
+
+##### Cosa devo fare?
+
+Rimuovi `extendDefaultPlugins: false` e aggiungi dei flag per disabilitare ciascun plugin indipendentemente.
+
+- `markdown.gfm: false` disables GitHub-Flavored Markdown
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:160]: **issue:** these 2 lines are still in English.
I would suggest:
Line 160: `disabilita GitHub-Flavored Markdown`
Line 161: `disabilita SmartyPants`

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+  ]
+});
+```
+
+##### Cosa devo fare?
+
+Rivedi le configurazioni di Markdown e MDX e confronta la tua configurazione con le nuove opzioni disponibili.
+
+
+### Cambiato: Accesso dei plugn all'avantesto
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:253]: **issue:** I think the word `avantesto` is out of context here (see e.g. https://www.treccani.it/vocabolario/avantesto/).
I would keep the english work `frontmatter` which is well understood in the context of markdown files.

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+
+Nella v1.x, i plugin remark e rehype non avevano accesso all'avantesto dell'utente. Astro ha unito l'avantesto del plugin con l'avantesto del tuo file, senza dover passare l'avantesto del file ai tuoi plugin.
+
+Astro v2.0 consente ai plugin remark e rehype l'accesso all'avantesto degli utenti tramite l'iniezione di avantesto. Ciò consente agli autori di plug-in di modificare l'avantesto esistente di un utente o di calcolare nuove proprietà basate su altre proprietà.
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:257]: **nitpick**: I would use `inserimento` instead of `iniezione`, it seems more fluent.

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 9]
```diff
@@ -0,0 +1,423 @@
...
+### Cambiato: Configurazione di RSS
+
+Nella v1.x, il pacchetto RSS di Astro permetteva di usare `items: import.meta.glob(...)` per generare un elenco di elementi del feed RSS. Questo utilizzo è ora obsoleto e alla fine sarà rimosso.
+
+Astro v2.0 introduce un `pagesGlobToRssItems()` wrapper alla proprietà `items`.
+
+
+#### Cosa devo fare?
+
+Importa, poi avvolgi la funzione esistente contenente `import.meta.glob()` con il helper `pagesGlobToRssItems()`.
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:274]: **nitpick:** I would rephrase this whole sentence to be more fluent.
E.g.

Importa la funzione helper `pagesGlobToRssItems()`, dopodiché passa la funzione esistente che utilizza `import.meta.glob()` come argomento all'helper stesso.

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 10]
```diff
@@ -0,0 +1,423 @@
...
+export default {
+  preprocess: vitePreprocess(),
+};
+```
+
+Per i nuovi utenti, questo file sarà aggiunto automaticamente con l'esecuzione di `astro add svelte`.
+
+### Rimosso: `legacy.astroFlavoredMarkdown`
+
+Nella v1.0, Astro ha spostato il vecchio Astro-Flavored Markdown (conosciuto anche come Components in Markdown) ad una vecchia funzionalità.
```
Comment (csarnataro) [inline src/content/docs/it/guides/upgrade-to/v2.mdx:312]: **nitpick:** I would rephrase this sentence to be more fluent.

Già nella v1.0, Astro aveva dichiarato come deprecato il vecchio Astro-Flavored Markdown (conosciuto anche come Components in Markdown).

File: src/content/docs/it/guides/upgrade-to/v2.mdx [hunk 11]
```diff
@@ -0,0 +1,423 @@
...
+
+```
+
+### Aggiornato: Vite 4
+
+Astro v2.0 passa da Vite 3 a [Vite 4](https://vitejs.dev/), rilasciato a Dicembre 2022.
+
+#### Cosa devo fare?
+
+Non dovrebbero essere necessarie modifiche al codice! Abbiamo gestito al tuo posto la maggior parte dell'aggiornamento all'interno di Astro; tuttavia, alcuni sottili comportamenti di Vite possono ancora cambiare tra le versioni.
```
[the remark this answers is not in the document]
Comment (akaik0) [reply src/content/docs/it/guides/upgrade-to/v2.mdx:394]: "some subtle Vite behaviors may" in this context subtle can't be translated as "leggermente differenti", I would suggest  "particolari" or "lievi".

**Review by csarnataro — APPROVED** [review]
LGTM

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks peeps 🚀
